granulate_utils/config_feeder/client/cluster_client.py

```diff
@@ -0,0 +1,71 @@
+import json
+import logging
+from dataclasses import dataclass, field
+from typing import Any, Dict, Optional, Union
+
+from granulate_utils.config_feeder.client.http_client import HttpClient
+from granulate_utils.config_feeder.client.models import CollectionResult
+from granulate_utils.config_feeder.core.models.aggregation import NodeResourceConfigCreate
+from granulate_utils.config_feeder.core.models.cluster import ClusterCreate, CreateClusterRequest, CreateClusterResponse
+from granulate_utils.config_feeder.core.models.collection import CollectorType
+from granulate_utils.config_feeder.core.models.node import CreateNodeRequest, CreateNodeResponse, NodeCreate, NodeInfo
+
+
+@dataclass
+class ClusterClient:
+    _logger: Union[logging.Logger, logging.LoggerAdapter]
+    _http_client: HttpClient
+    _collector_type: CollectorType
+    _service: str
+    _cluster_id: Optional[str] = field(init=False, default=None)
+
+    def register_cluster_if_needed(self, node_info: NodeInfo, collection_result: Dict[str, CollectionResult]) -> None:
+        if self._cluster_id is None and (node_info.is_master or collection_result):
+            self._register_cluster(node_info)
+
+    def submit_node_configs(
+        self,
+        node_info: NodeInfo,
+        config_requests: Dict[str, NodeResourceConfigCreate],
+    ) -> Dict[str, Any]:
+        node_id = self._register_node(node_info)
+        self._logger.info(f"sending configs for node {node_info.external_id}")
+        request = {}
```

Comment:
Not great, but not terrible - the alternative is to somehow use Pydantic's `create_model` and create this model in runtime, not sure it's worth it.